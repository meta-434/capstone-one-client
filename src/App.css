@import url('https://rsms.me/inter/inter.css');
  html { font-family: 'Inter', sans-serif; }
@supports (font-variation-settings: normal) {
  html { font-family: 'Inter var', sans-serif; }
}
@supports (font-variation-settings: normal) {
  html { font-family: 'Inter Bold', sans-serif; }
}

* {
  box-sizing: border-box;
  font-family: 'Inter';
  background: url('images/background.png') no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

body {
  font-size: 20px;
  color: #444;
  text-align: center;
  line-height: 1.5;
}

nav,
footer {
  padding: 1% 0;
}

.error {
  margin-top: 140px;
}

nav {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
  height: 70px;
}

header {
  /*padding: 45px 15px;*/
  margin-top: 70px;
}

section {
  min-height: 200px;
  padding: 30px 15px;
  color: #5E81AC;
}

p {
  margin: 0 auto;
  max-width: 650px;
}

p + p {
  margin-top: 15px;
}

button:disabled {
  border: 2px #4C566A solid;
  border-radius: 3px;
  color: #4C566A;
}

button {
  border: 2px inherit solid;
  border-radius: 3px;
}