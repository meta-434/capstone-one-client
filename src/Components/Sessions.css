header {
    font-size: 40px;
    min-height: 100px;
    padding-top: 20px;
}

.sessions-section > h2 {
    margin-top: 70px;
}

section > p {
    margin-top: 0;
}

.sessions-display {
    margin-top: 0;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

.sessions-display > div {
    margin: 0.5%;
    text-align: center;
    min-width: 200px;
    width: 16.5%;
    max-height: 400px;
    min-height: 400px;
    border: 2px #5e81ac solid;
    border-radius: 3px;
    flex-direction: column;
}

.sessions-display > div > h4 {
    overflow: auto;
}

.sessions-display > button {
    border: 2px #5e81ac solid;
    margin-bottom: 10px;
    color: #5e81ac;
    font-size: 12px;
}