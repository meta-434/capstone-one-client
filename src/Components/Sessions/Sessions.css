header {
    font-size: 40px;
    min-height: 100px;
    padding-top: 20px;
}

.sessions-section > h2 {
    margin-top: 70px;
}

section > p {
    margin-top: 0;
}

.sessions-display {
    margin-top: 0;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
}

.sessions-display > div {
    margin: 0.5%;
    text-align: center;
    min-width: 200px;
    width: 16.5%;
    max-height: 400px;
    min-height: 400px;
    border: 2px #8DA5C4 solid;
    border-radius: 3px;
    flex-direction: column;
    flex-wrap: nowrap;
}

.sessions-display > div > h4, .sessions-display > div > h3 {
    overflow: scroll;
    word-break: break-all;
}

.sessions-display > div > h3 {
    min-height: 40px;
}

.sessions-display > button {
    border: 2px #8DA5C4 solid;
    margin-bottom: 10px;
    color: #8DA5C4;
    font-size: 12px;
}

@media only screen and (max-width: 600px){
    .sessions-display {
        flex-direction: column;
    }

    .sessions-display > div {
        margin-top: 10px;
    }
}