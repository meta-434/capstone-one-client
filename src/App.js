import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Nav from './Components/Nav/Nav.js';
import Login from './Components/Login/Login.js';
import Signup from './Components/Signup/Signup.js';
import Main from './Components/Main/Main.js';
import Notes from './Components/Notes/Notes.js';
import Sessions from './Components/Sessions/Sessions.js';
import TimerComponent from './Components/TimerComponent/TimerComponent.js';
import PomodoroContext from './PomodoroContext';
import './App.css'

class App extends Component {
  state = {
    sessions: [],
    notes: [],
    authToken: undefined,
    error: undefined,
  }

  logOut = () => {
    this.setState({authToken: undefined});
    sessionStorage.clear();
  }

  handlePostSignup = ({ username, password }) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/signup`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        'username': username,
        'password': password,
      })
    })
        .then(result => {
            let json = result.json() // There's always a response body
            if (result.status >= 200 && result.status < 300) { return json }
            return json.then(Promise.reject.bind(Promise))

        })
        .then(result =>  result.json())
        .catch(error => {
          this.setState({error});
        });
  };

    handlePostAuthenticate = ({ username, password }) => {
        console.log('for login');
        fetch(process.env.REACT_APP_SERVER_URL + `/authenticate`, {
            method: 'post',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                'username': username,
                'password': password,
            })
        })
            .then(result => {
                let json = result.json() // There's always a response body
                if (!(result.status >= 200 && result.status < 300)) {
                    return json.then(Promise.reject.bind(Promise))
                }
                return json;
            })
            .then(resJson => {
                if (!!resJson.token) {
                    this.setState({ authToken: resJson.token, username })
                    sessionStorage.setItem('access-token', this.state.authToken);
                    sessionStorage.setItem('username', username);
                    this.handleGetSessions();
                    this.handleGetNotes();
                }
                else {
                    throw new Error(' error in authenticating. check username and password. ');
                }
            })
            .catch(error => {
                console.error(error);
                this.setState({error});
            });
    };

  handlePostSession = ({session_name, session_description, ownerId}) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/sessions`, {
      method: 'post',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      },
      body: JSON.stringify({
        session_name,
        session_description,
        session_owner: ownerId
      })
    })
        .then(res => this.handleGetSessions())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleGetSessions = () => {
    fetch(process.env.REACT_APP_SERVER_URL+ `/api/sessions`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(response => response.json())
        .then(responseJson => {
              if (responseJson.success && responseJson.success === false) {
                throw new Error('error in getting sessions')
              } else {
                this.setState({
                  sessions: responseJson
                })
              }
            }
        )
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleDeleteSession = (id) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/sessions/${id}`, {
      method: 'delete',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(res => this.handleGetSessions())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handlePostNote = ({note_name, note_content, ownerId}) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/notes`, {
      method: 'post',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      },
      body: JSON.stringify({
        note_name,
        note_content,
        note_owner: ownerId
      })
    })
        .then(res => this.handleGetNotes())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleGetNotes = () => {
    fetch(process.env.REACT_APP_SERVER_URL+ `/api/notes`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(response => response.json())
        .then(responseJson => {
              if (responseJson.success && responseJson.success === false) {
                throw new Error('error in getting notes')
              } else {
                this.setState({
                  notes: responseJson
                })
              }
            }
        )
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleDeleteNote = (id) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/notes/${id}`, {
      method: 'delete',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(res => this.handleGetNotes())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  render () {
    const { sessions, notes, authToken, username, error } = this.state;
    const context = {
      sessions,
      notes,
      username,
      authToken,
      error,
      handlePostAuthenticate: this.handlePostAuthenticate,
      handlePostSignup: this.handlePostSignup,
      logOut: this.logOut,
      handleGetNotes: this.handleGetNotes,
      handlePostNote: this.handlePostNote,
      handleDeleteNote: this.handleDeleteNote,
      handleGetSessions: this.handleGetSessions,
      handleDeleteSession: this.handleDeleteSession,
      handlePostSession: this.handlePostSession,
    }
    return (
        <PomodoroContext.Provider value={(context)}>
          <div className="app">
            <Route path="/" render={routeProps => <Nav {...routeProps}/>}/>
            <main className="app-content">
              <Route exact path="/" component={Main} />
              <Route path="/login" component={Login}/>
              <Route path="/signup" component={Signup} />
              <Route path="/timer" component={TimerComponent}/>
              <Route path="/sessions/:sessionId?" component={Sessions}/>
              <Route path="/notes/:noteId?" component={Notes}/>

            </main>
          </div>
        </PomodoroContext.Provider>
    );
  }
}

export default App;