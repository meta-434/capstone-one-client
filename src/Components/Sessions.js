import React, { Component } from 'react';
import './Sessions.css'

class Sessions extends Component {
    render() {
        return(
            <section className="sessions-section">
                <h2>sessions placeholder</h2>
                <p>hello?</p>
            </section>
        );
    }
}

export default Sessions;