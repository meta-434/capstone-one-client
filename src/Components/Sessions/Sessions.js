import React, { Component } from "react";
import PomodoroContext from "../../PomodoroContext";
import "./Sessions.css";
import ErrorDisplay from "../ErrorDisplay/ErrorDisplay";

class Sessions extends Component {
  static contextType = PomodoroContext;

  handleDelete = (id) => {
    this.context.handleDeleteSession(id);
  };

  componentDidMount() {
    this.context.handleGetSessions();
  }

  render() {
    return (
      <>
        {!!this.context.error ? <ErrorDisplay /> : ""}
        <section>
          <header>My Saved Sessions</header>
          <h3>review past work sessions</h3>
        </section>
        <section className="sessions-display">
          <PomodoroContext.Consumer>
            {({ sessions }) => {
              if (sessions.length !== 0 && sessions.success !== false) {
                return sessions
                  .map((session, index) => {
                    return (
                      <div
                        className="sessions-display"
                        key={index}
                        id={session.id}
                      >
                        <h3>{session.session_name}</h3>
                        <h4>â†³ {session.session_description}</h4>
                        <p>session completed: {session.session_end}</p>
                        <button onClick={() => this.handleDelete(session.id)}>
                          Delete
                        </button>
                      </div>
                    );
                  })
                  .reverse();
              }
            }}
          </PomodoroContext.Consumer>
        </section>
      </>
    );
  }
}

export default Sessions;
