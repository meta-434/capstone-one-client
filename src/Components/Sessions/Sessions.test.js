import React from 'react';
import { render } from '@testing-library/react';
import Sessions from './Sessions';
import App from '../../App';
import {BrowserRouter} from "react-router-dom";
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PomodoroContext from "../../PomodoroContext";

test('renders Sessions component without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(
        <PomodoroContext.Provider value={{
            sessions: [
                {
                    session_id: 1,
                    session_name: 'jest session',
                    session_content: 'jest test session',
                    session_owner: 30,
                },
            ],
            handleGetSessions: () => {
                return {
                    session_name: 'jest session',
                    session_content: 'jest session contents',
                    session_owner: 1
                };
            }
        }}>
            <Sessions />
        </PomodoroContext.Provider>
        , div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the Sessions UI as expected', () => {
    const tree = renderer
        .create(
            <PomodoroContext.Provider value={{
                sessions: [
                    {
                        session_id: 1,
                        session_name: 'jest session',
                        session_description: 'jest test session',
                        session_end: new Date().getMonth(),
                        session_owner: 30,
                    },
                ],
                handleGetSessions: () => {
                    return {
                        session_name: 'jest session',
                        session_description: 'jest session description',
                        session_end: new Date().getMonth(),
                        session_owner: 1
                    };
                }
            }}>
                <Sessions />
            </PomodoroContext.Provider>

        )
        .toJSON();

    expect(tree).toMatchSnapshot();
})