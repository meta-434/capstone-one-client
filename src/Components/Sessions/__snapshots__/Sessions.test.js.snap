// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Sessions UI as expected 1`] = `
Array [
  "",
  <section>
    <header>
      My Saved Sessions
    </header>
    <h3>
      review past work sessions
    </h3>
  </section>,
  <section
    className="sessions-display"
  >
    <div
      className="sessions-display"
    >
      <h3>
        jest session
      </h3>
      <h4>
        â†³ 
        jest test session
      </h4>
      <p>
        session completed: 
        4
      </p>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </section>,
]
`;
